libftl/media.c

```diff
@@ -586,34 +582,49 @@ static int _nack_resend_packet(ftl_stream_configuration_private_t *ftl, uint32_t
 	return tx_len;
 }
 
+static int _write_rtp_header(uint8_t *buf, size_t len, uint8_t ptype, uint16_t seq_num, uint32_t timestamp, uint32_t ssrc) {
+	uint32_t rtp_header;
+	
+	if (RTP_HEADER_BASE_LEN > len) {
+		return -1;
+	}
+
+	//TODO need to worry about alignment on some platforms
+	uint32_t *out_header = (uint32_t *)buf;
+
+	rtp_header = htonl((2 << 30) | (ptype << 16) | seq_num);
+
+	*out_header++ = rtp_header;
+	rtp_header = htonl((uint32_t)timestamp);
+	*out_header++ = rtp_header;
+	rtp_header = htonl(ssrc);
+	*out_header++ = rtp_header;
```

Comment:
packet buffer is 32bit (or better) aligned so this isn't necessary in this case.  most modern processors support unaligned memory accesses albiet with a bit of a penalty but often better than byte accesses.  for an extreme level of portability it's a valid point